src/components/customers/subscriptions/EditCustomerSubscriptionDrawer.tsx

```diff
@@ -0,0 +1,123 @@
+import { forwardRef, useRef, useState, useImperativeHandle } from 'react'
+import { useFormik } from 'formik'
+import { gql } from '@apollo/client'
+
+import { Button, DrawerRef, Drawer, Typography } from '~/components/designSystem'
+import { useInternationalization } from '~/hooks/core/useInternationalization'
+import { TextInputField, DatePickerField } from '~/components/form'
+import { addToast } from '~/core/apolloClient'
+import {
+  useUpdateCustomerSubscriptionMutation,
+  UpdateSubscriptionInput,
+  StatusTypeEnum,
+} from '~/generated/graphql'
+import { Card, DrawerTitle, DrawerContent, DrawerSubmitButton } from '~/styles'
+
+gql`
+  mutation updateCustomerSubscription($input: UpdateSubscriptionInput!) {
+    updateSubscription(input: $input) {
+      id
+      name
+    }
+  }
+`
+
+type SubscriptionInfos = {
+  id: string
+  name?: string | null
+  startDate: string
+  status?: StatusTypeEnum
+}
+
+export interface EditCustomerSubscriptionDrawerRef {
+  openDrawer: (subscriptionInfos: SubscriptionInfos) => unknown
+  closeDrawer: () => unknown
+}
+
+export const EditCustomerSubscriptionDrawer = forwardRef<EditCustomerSubscriptionDrawerRef>(
+  (_, ref) => {
+    const drawerRef = useRef<DrawerRef>(null)
+    const [update] = useUpdateCustomerSubscriptionMutation({
+      onCompleted({ updateSubscription }) {
+        if (!!updateSubscription) {
+          addToast({
+            severity: 'success',
+            translateKey: 'text_62e38a4631937146a6d6d5dd',
+          })
+        }
+      },
+    })
+    const [subscription, setSubscription] = useState<SubscriptionInfos | undefined>(undefined)
+    const { translate } = useInternationalization()
+    const formikProps = useFormik<Pick<UpdateSubscriptionInput, 'name' | 'subscriptionDate'>>({
+      initialValues: {
+        name: subscription?.name || '',
+        subscriptionDate: subscription?.startDate || undefined,
+      },
+      validateOnMount: true,
+      enableReinitialize: true,
+      onSubmit: async (values) => {
+        await update({
+          variables: {
+            input: {
+              id: subscription?.id as string,
+              ...values,
+            },
+          },
+        })
+      },
+    })
+
+    useImperativeHandle(ref, () => ({
+      openDrawer: (infos) => {
+        formikProps.resetForm()
```

Comment:
Question, are the validations are also reset if a previous drawer have been opened ? Asking because I had issues with that in modals recently

Reply:
For some reason, yes it does 🫠